build/common.sh

```diff
@@ -501,9 +501,11 @@ function kube::build::ensure_data_container() {
   # If the data container exists AND exited successfully, we can use it.
   # Otherwise nuke it and start over.
   local ret=0
-  local code=$(docker inspect \
+  local code=0
+
+  code=$(docker inspect \
       -f '{{.State.ExitCode}}' \
-      "${KUBE_DATA_CONTAINER_NAME}" 2>/dev/null || ret=$?)
+      "${KUBE_DATA_CONTAINER_NAME}" 2>/dev/null) || ret=$?
   if [[ "${ret}" == 0 && "${code}" != 0 ]]; then
```

Comment:
Current statement is a bit convoluted, the following might be clearer?

```sh
if code="$(docker inspect -f '{{.State.ExitCode}}' "${KUBE_DATA_CONTAINER_NAME}")" && [[ "${code}" != 0 ]]; then
```

Also to note the assignment followed by a `||` has a potential bug where `code` has been declared `read-only`, so the _assignment_ of a value to `code` will fail even if the subshell that generates the value does not, so in either case there are failure conditions where `docker inspect` exits non-zero and the container also exited cleanly.

Reply:
Note that errors in the subshell fail the whole `if` block:


```sh
$ if [[ "$( grep 2>&1 )" == "test" ]]; then echo success; fi
$
```

Reply:
Thanks for the review, @stevekuznetsov!

I think meaning of the code is different in your example. Currently, if the `docker inspect` command succeeds and the container internal exit code is 0, then the data container is not destroyed and also it's not created again.

In your code, this case would mean that the `if` block fails because the container internal exit code would be 0. At that point we wouldn't have anymore the information whether the failure was due to the container exit code being 0 (when nothing would need to be done) and for `docker inspect` failing to find the container (in which case the container would need to be recreated).

Reply:
Not sure I follow, in my snippet I was still checking `!= 0`. In any case, it's a minor nit.